src/wallet/wallet2.cpp

```diff
@@ -889,14 +927,16 @@ void wallet2::expand_subaddresses(const cryptonote::subaddress_index& index)
     cryptonote::subaddress_index index2;
     for (index2.major = m_subaddress_labels.size(); index2.major < index.major + m_subaddress_lookahead_major; ++index2.major)
     {
-      for (index2.minor = 0; index2.minor < (index2.major == index.major ? index.minor : 0) + m_subaddress_lookahead_minor; ++index2.minor)
+      uint32_t end = (index2.major == index.major ? index.minor : 0) + m_subaddress_lookahead_minor;
+      index2.minor = 0;
+      while (m_subaddresses_inv.count(index2) != 0)
```

Comment:
The logic design seems to imply that this while-loop is never executed? Not a big deal, though.

Reply:
This is to match the if (m_subaddresses_inv.count(index2) == 0), which implied the labels and subaddresses could be unmatched (if an address has no label I guess). Admittedly only at start, but the overwrite of an existing should be a no-op. If you think this can't happen I'll remove it, it's just an optimization to avoid calculating already known data.

Reply:
Ah, looks like the original scheme using `m_subaddresses_inv` was redundant and unnecessary altogether! I think we can safely remove it and eliminate this no-op checking, making the wallet cache smaller. Would you mind doing that here?

Reply:
OK